feature-libs/order/components/order-details/order-details-extended/consignment-tracking/consignment-tracking-link.component.ts

```diff
@@ -5,56 +5,28 @@
  */
 
 import { Component, HostBinding, inject, OnInit } from '@angular/core';
-import {
-  LaunchDialogService,
-  LAUNCH_CALLER,
-  OutletContextData,
-} from '@spartacus/storefront';
-import { Observable, of } from 'rxjs';
-import { map, take } from 'rxjs/operators';
-import { Consignment } from '@spartacus/order/root';
-import { OrderConsignmentsService } from '../../order-consignments.service';
-
+import { Consignment, Order } from '@spartacus/order/root';
+import { OutletContextData } from '@spartacus/storefront';
+import { ConsignmentTrackingComponent } from '../../order-detail-items';
+type ConsignmentOutletContextData = { item: Consignment; order?: Order };
 @Component({
   selector: 'cx-consignment-tracking-link',
   templateUrl: './consignment-tracking-link.component.html',
-  // host: { class: 'cx-list-header col-12' },
 })
-export class ConsignmentTrackingLinkComponent implements OnInit {
-  @HostBinding('className') componentClass: string;
-  consignment$: Observable<Consignment>;
-  protected orderConsignmentsService = inject(OrderConsignmentsService);
-  protected launchDialogService = inject(LaunchDialogService);
+export class ConsignmentTrackingLinkComponent
+  extends ConsignmentTrackingComponent
+  implements OnInit
+{
+  @HostBinding('className') componentClass: string = 'cx-list-header col-12';
   protected outlet = inject(OutletContextData);
 
   ngOnInit(): void {
-    this.componentClass = 'cx-list-header col-12';
-    this.consignment$ = this.outlet?.context$.pipe(
-      map((context) => context.item)
-    );
-  }
-  openTrackingDialog(consignment: Consignment) {
-    let tracking = {};
-    if (consignment.consignmentTracking) {
-      tracking = consignment.consignmentTracking;
-    }
-    const modalInstanceData = {
-      tracking$: of(tracking),
-      shipDate: consignment.statusDate,
-    };
-
-    const dialog = this.launchDialogService.openDialog(
-      LAUNCH_CALLER.CONSIGNMENT_TRACKING,
-      undefined,
-      undefined,
-      modalInstanceData
-    );
-
-    if (dialog) {
-      dialog.pipe(take(1)).subscribe();
-    }
-  }
-  isStatusValid(status: string): boolean {
-    return this.orderConsignmentsService.isStatusValid(status);
+    this.outlet?.context$.subscribe((context: ConsignmentOutletContextData) => {
+      this.consignment = context.item;
+    });
+    this.outlet?.context$.subscribe((context: ConsignmentOutletContextData) => {
+      this.orderCode = context?.order?.code ?? '';
+    });
+    super.ngOnInit();
```

Comment:
[MEDIUM]
Why we got back to copying observable values to static properties?

If it's only to be able to inherit from the class `ConsignmentTrackingComponent`... is inheriting from this class bringing really enough value to justify the mixed approach (observables and static properties)?

Reply:
since we are extending already existing component , it uses static variable, The parent component holds the logic to show dialog.